internal/ir/xds.go

```diff
@@ -7,9 +7,20 @@ package ir
 
 import (
 	"errors"
+	"fmt"
 	"net"
 
 	"github.com/tetratelabs/multierror"
+
+	egv1a1 "github.com/envoyproxy/gateway/api/v1alpha1"
+	"github.com/envoyproxy/gateway/api/v1alpha1/validation"
+)
+
+const (
+	maxJwtRules          = 4
```

Comment:
if these are user facing limits, shouldnt they be enforced in the provider or resource translator instead ?

Reply:
@arkodg these limits, e.g. validation rules, are specified in the [AuthenticationFilter API](https://github.com/envoyproxy/gateway/blob/main/api/v1alpha1/authenticationfilter_types.go) and will be enforced by the kube-api server. The validation rules are added to the xds IR for consistency and additional protection. I can remove them if you have a strong opinion against this approach.